Peanuts and peanut butter! + Other various new foods

MidoriWroth opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

This PR will:

- Add growable peanuts! They can be grown in botany and eaten as is, or ground into peanut butter to be used for various other recipes. Peanut butter can also be ordered at the order console for 30 credits per jar.
- Add a few new peanut related recipes, including: Peanut butter and jelly sandwich, peanut butter and banana sandwich, peanut butter cookies, and granola bars.
- Add brownies! A treat that has been long neglected from the game. Initially the batter is made, which must be baked in an oven and then sliced into pieces.
- Add breadsticks! A simple side to make for many dishes.
- Add nigiri sushi, for a little more sushi variety as per many requests.
- Changed the meatball noodle soup to require peanuts instead of Gallery's Peanuts

![image](https://user-images.githubusercontent.com/18170896/178155835-13f40307-3582-42a1-abdb-37aed3a241ed.png)


## Why It's Good For The Game

More foods add more variety to the possible menus chefs can make. I was surprised that peanuts and brownies were not a thing in the game yet, so I had to gadd them

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
add: Added growable peanuts! Grow them in botany for a snack, or grind them into peanut butter
add: Added 4 different peanut and peanut butter-based recipes
add: Added brownies. Yes, I know, finally
add: Added breadsticks, as if food addiction couldn't get worse
add: Added a simple nigiri sushi, for a little more sushi variety. Expect more in the future.
qol: Meatball noodle soup now requires grown peanuts instead of Gallery's peanuts
imageadd: New sprites for all the new foods!
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

MidoriWroth committed:
Food expansion 3!

newfren commented:
there's a bag of peanuts in the junk food vendors that's used in a couple recipes

Wallemations commented:
Now all I need is celery and we can make literal ants on a log

Commit:
dehardcodes admin grantable objectives list (#68245)

* Update objective.dm

* Update objective.dm

* Update objective.dm

* Update code/game/gamemodes/objective.dm

Co-authored-by: John Willard <53777086+JohnFulpWillard@users.noreply.github.com>

* Update objective.dm

Co-authored-by: John Willard <53777086+JohnFulpWillard@users.noreply.github.com>

tgstation-server committed:
Automatic changelog generation for PR #68245 [ci skip]

StrangeWeirdKitten committed:
Fixes Bubblegum (#68266)

* Update bubblegum.dm

* Update code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum.dm

Co-authored-by: ShizCalev <ShizCalev@users.noreply.github.com>

* Ok

Ok

Co-authored-by: ShizCalev <ShizCalev@users.noreply.github.com>

tgstation-server committed:
Automatic changelog generation for PR #68266 [ci skip]

EOBGames commented on code/modules/reagents/chemistry/reagents/food_reagents.dm:
```diff
@@ -1063,3 +1063,10 @@
 	color = "#adcf77"
 	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
 
+/datum/reagent/consumable/peanut_butter
```
peanut butter is already a reagent, added in #59412 - it's a few above this new define in the code, specifically between whipped cream and vinegar, so adding this is unnecessary

GoldenAlpharex commented on code/game/objects/items/food/bread.dm:
```diff
@@ -313,8 +313,32 @@
 	w_class = WEIGHT_CLASS_SMALL
 	burns_on_grill = TRUE
+
+/obj/item/food/rawbreadstick
```
Always use `snake_case` whenever you can, when writing typepaths and variable names.

There's a lot of them so I won't comment on each of them, but if you need help figuring it out, hit me up.
```suggestion
/obj/item/food/raw_breadstick
```

MidoriWroth committed:
touched up some sprites + fixed syntax

MidoriWroth commented:
yeah idk how i didn't realize that peanut butter was already a thing lmao, played this game for years and never once saw it.

YakumoChen committed:
Makes GPS units autolathe banned (#68100)

* Bans GPS' from being added to Autolathes' Imported section.

tgstation-server committed:
Automatic changelog generation for PR #68100 [ci skip]

Sucy-Manbavaran commented:
Looking at the code, im a bit inspired on adding some new foods! i didn't realize how simple it could be ahaha, though i see a lot of places where i could make mistakes! And i do hope this gets merged! these are some very fun sounding recipes and i'd love to be able to make them!

san7890 commented:
![image](https://user-images.githubusercontent.com/34697715/178124882-5dfe3f7b-2a82-4db5-a1b7-a2d67a34759d.png)

The bacon updates aren't mentioned in your PR (via IconDiffBot2), is this intentional? It also appears you might be reverting #68255 in regards to the mint.

MidoriWroth commented:
> ![image](https://user-images.githubusercontent.com/34697715/178124882-5dfe3f7b-2a82-4db5-a1b7-a2d67a34759d.png)
> 
> The bacon updates aren't mentioned in your PR (via [IconDiffBot2](https://github.com/tgstation/tgstation/pull/68289/checks?check_run_id=7266185312)), is this intentional? It also appears you might be reverting #68255 in regards to the mint.

oh! uhhh, no, that was not intentional. let me see if i can fix it

MidoriWroth committed:
dunno why this was changed hehe

MidoriWroth committed:
lets not forget the mint too =)

actions-user committed:
Automatic changelog compile [ci skip]

MidoriWroth commented:
> Looking at the code, im a bit inspired on adding some new foods! i didn't realize how simple it could be ahaha, though i see a lot of places where i could make mistakes! And i do hope this gets merged! these are some very fun sounding recipes and i'd love to be able to make them!

yeah it's very simple! i'd like to see more stuff added, so go for it!

MidoriWroth committed:
new peanut butter jar sprite

MidoriWroth commented:
![image](https://user-images.githubusercontent.com/18170896/178131893-e8e8e55b-1e20-47c6-a824-9246e8230257.png)

New peanut butter jar sprite courtesy of @ArcaneMusic who is the GOAT

san7890 committed:
Fixes Gulag Shuttle Access On Tramstation (#68221)

* Fixes Gulag Shuttle Access On Tramstation

Hey there,

The prisoners actually enter on the LEFT, not the right, so these accesses were not actually set properly. Let's fix that up right now.

* Revert "Fixes Gulag Shuttle Access On Tramstation"

This reverts commit dd1b1c25ad5351cdba515c2f0d77494f409f7ed2.

* ok hard reset and redo

tgstation-server committed:
Automatic changelog generation for PR #68221 [ci skip]

MrMelbert committed:
Fixes Tonic Water causing dizziness (#68276)

Fixes tonic water causing dizziness

tgstation-server committed:
Automatic changelog generation for PR #68276 [ci skip]

san7890 committed:
Fixes TramStation's Xenobiology Disposals (#68286)

tgstation-server committed:
Automatic changelog generation for PR #68286 [ci skip]

MidoriWroth committed:
Oopsies forgot to update icons too teehee

san7890 committed:
[s] Patches potential runtimes in AdminPM (#68324)


About The Pull Request

Hey there,

6e29689 seemed to work alright on my end, but apparently there are cases where current_ticket.id = NULL. This would cause runtime trace errors that would start to list out the first few words of most adminhelps. Fuck.

You probably could use recipient.current_ticket.id on the "else" portion but I woke up to this and I'm not 100% certain if that will null out at this time. Might also be apt to revert #68215 entirely?
Why It's Good For The Game

Those shouldn't show up in runtimes!
Changelog

Nothing particularly player facing.

LemonInTheDark committed:
Refactors mothweek check (#68280)

* Refactors mothweek check

Mothweek HAS SIDE EFFECTS
Checking it on a day that causes a week offset will currently cause that
week offset to persist
That's dumb.

Also it's not very expansive, only covers a slim set of possibilities.

Instead, lets build something to generate all passing days over a period
of time, maybe 3 months out of 2 years.
Then we'll crosscheck that against some predecided "ok" dates

If either list disagrees with each other, we'll fail. That way we can't
miss an edgecase. or have issues with side effects

I like this pattern.

tf-4 commented:
That peanut butter jar seems a bit uh, huge, as good as the sprite is

actions-user committed:
Automatic changelog compile [ci skip]

MidoriWroth commented:
i'll see if i can make it a bit smaller

Commit:
[NO GBP] changes box storage limit to 2 instead of 4 (#68298)

fun ruined

tgstation-server committed:
Automatic changelog generation for PR #68298 [ci skip]

13spacemen committed:
Shortens Hub Entry Descriptors (#68186)

Timberpoes committed:
Removes temporary debugging code that can spam admin chat. (#68321)

Feex

tgstation-server committed:
Automatic changelog generation for PR #68321 [ci skip]

MidoriWroth committed:
Ok updated peanut butter container again

MidoriWroth commented:
so i made this one, quite pleased with it, it's somewhat of a smaller sprite 
![image](https://user-images.githubusercontent.com/18170896/178211351-c7e01a0b-8686-4374-b3b7-19f287114a26.png)

MidoriWroth committed:
oopsies forgot this icon too

san7890 committed:
Shady Backroom Deals - Bartender's Area Definition (#68311)

add: bartender's cubby area is now known as the "backroom"

tgstation-server committed:
Automatic changelog generation for PR #68311 [ci skip]

LemonInTheDark committed:
Converts inappropriate update_overlays -> update_appearance (#68315)

IT DOESN'T DO ANYTHING
update_overlays is a helper proc, it returns a list of overlays to
add/track.
It doesn't update anything on its own.

Kapu1178 committed:
Ambience tweaks (#68264)

* cleans up code and adjusts volume

* Removes can_hear() check

* volume

* review

tgstation-server committed:
Automatic changelog generation for PR #68264 [ci skip]

LemonInTheDark committed:
Prevents overtime in the economy SS (#68078)

* Prevents overtime in the economy SS

Ok so like, we were doing a LOT of work all at once without yielding,
which is fine cause like, it's only fired once every 5 minutes, but it's
still kinda dumb

So I've cleaned up the subsystem to make it support an ssair style of
yielding, alongside an optimization to vending machine z level checks
that I suspect was taking a lot of cpu.

Remind me to refactor z level lookup someday bestie

Anyway yeah now we won't see strangely high overtime for ssecon anymore

* vending gang (renames a variable to not just be v

* It is done

Commit:
[NO GBP] properly updates the parents icon when an item is added to storage (#68305)

tgstation-server committed:
Automatic changelog generation for PR #68305 [ci skip]

RimiNosha committed:
Un-headache-ify Lavaland's Ambiance! (#68323)

tgstation-server committed:
Automatic changelog generation for PR #68323 [ci skip]

Couls committed:
Fixes for AI detector (#68301)

Fixes multitool detection

Also modifies the uplink description to reflect the changes made to the multitool and makes the HUD mode into toggleable detection mode, and removes the unused hud code

tgstation-server committed:
Automatic changelog generation for PR #68301 [ci skip]

Commit:
[NO GBP] fixes issue where observers couldn't open storage items (#68299)

* fun ruined

* shit

* woops

tgstation-server committed:
Automatic changelog generation for PR #68299 [ci skip]

zxaber committed:
Combat mech research nodes now require mech-themed experiments rather than bomb experiments. (#67756)

* Mechs require a new Mech scanning experiment centered around damaging/destroying mechs, instead of being behind weaponry, which requires ordnance research.

tgstation-server committed:
Automatic changelog generation for PR #67756 [ci skip]

Salex08 committed:
Makes silver spawned food actually disgusting (#68319)

Fixes silver spawned food not tasting disgusting by sending a signal to the Edible component to make it disgusting.

tgstation-server committed:
Automatic changelog generation for PR #68319 [ci skip]

san7890 committed:
Adds an Air Alarm to TramStation Bar Lounge (#68308)

tgstation-server committed:
Automatic changelog generation for PR #68308 [ci skip]

actions-user committed:
Automatic changelog compile [ci skip]

Spookuni committed:
Gives Delta genetics a "back-desk" to encourage interaction (#68340)

Gives Delta genetics a basic desk leading into the science maint area to make interacting with them a little bit easier.

Having open desks makes it less likely that people succumb to the siren song of being a job-completing gremlin who never talks to anyone else, interacts with them or uses their job content to benefit the station as a whole. A fully public front desk would be preferable for this, but it would require a major rework of the science department that is beyond my meagre mapping abilities. Letting geneticists more easily interact with passing assistants/secoffs/parameds/engineers/anyone with maint access will hopefully help a little in the fight against job-gremlinism.

tgstation-server committed:
Automatic changelog generation for PR #68340 [ci skip]

unit0016 committed:
Icebox Murals: AI Upload and RD Office (#68341)

Redoes the Icebox upload slightly. Adds murals to the upload and RD's office. Also compresses the tile decals on the bridge.

Murals are fun ways to implement art into spaces, whether they commune something or are just there to look nice. It's weird we don't have any in-game already.

tgstation-server committed:
Automatic changelog generation for PR #68341 [ci skip]

jlsnow301 committed:
Fixes admin.fun list (#68336)


About The Pull Request

Many of the items in the list of admin fun were not actually categorized as Admin.Fun (and thus not appearing), and some admin fun procs were not even listed.
Why It's Good For The Game

Basically, more admin fun procs are available that were already in code but not listed
Changelog

cl
admin: Fixed many "admin.fun" listings that were incorrect or missing.
/cl

tgstation-server committed:
Automatic changelog generation for PR #68336 [ci skip]

13spacemen committed:
Removes useless colon in hub entry (#68347)

Removes unnecessary colon in hub entry.

ReinaCoder committed:
Resprites the safe and floor safe (#68284)

imageadd: The safe and floor safes have been resprited

tgstation-server committed:
Automatic changelog generation for PR #68284 [ci skip]

EOBGames committed:
Poster Pack 2 - Beyond Posters (#68235)

Adds a new selection of posters, poster-like objects, and wall decorations.

New entirely with this PR, shipping containers! These sprites were adapted from a base from TGMC, with brands and designs more fitting to our universe here on TG

Another new addition, flags! These draw upon Common Core lore, representing several major factions from the lorebase as a new mapping tool

Finally, there are two interactive wall objects- wall clocks, and calendars:

The wall clock shows both the "real time", represented as the Earth time, or Treaty Coordinated Standard Time, and the station time, represented as the local time, or Coalition Standard Time

The calendar gives the in-universe date, as well as any active holidays

Why It's Good For The Game

This adds a lot of fun aesthetic additions that I think will benefit our mapping team in creating new spaces and breathing new life into older ones. Additionally, it helps make our Common Core lore a bit more tangible to the playerbase, bringing it away from being "just on a wiki" and into the game proper.

A whole host of new posters have just become available.
For the architects among you: a number of new aesthetic options are available for mapping with. Let your creativity free!

tgstation-server committed:
Automatic changelog generation for PR #68235 [ci skip]

Thunder12345 committed:
Mining Outpost Overhaul: Island Living Edition (#67796)

Replaced the old mining outpost with a new base consisting of separate buildings on islands in the middle of lava. Added a few new areas and decorations to fill out the outpost. Changed the underfloor accessibility of lava so cables and pipes don't mysteriously disappear as they cross catwalks over lava.

The current mining station is fairly boring and dated. In addition, the general layout of the outpost was bland and uninspired, about as exciting as any other hallway. This replacement offers a new and unique take with an interesting layout and new dangers to be aware of.

The bridges between buildings are exposed to the weather, meaning you could find yourself forced to choose between waiting out a storm in an isolated building with a potentially unfriendly roommate, or running through the storm to get back to a shuttle.

Power and communications are also more vulnerable in the new base, placed in a building on the mainland where ne'er-do-wells could gain access unnoticed and sabotage them, making the mining outpost less of a safe space with little risk of anything going seriously wrong. You don't get access to these just for having mining access, better call the relevant portion of engineering to help you.

Gulag prisoners also get a few new options to keep security on their toes, with a few different options for escape routes that encourage the warden to keep an eye on their charges instead of ignoring the gulag while waiting for the prisoners to do their sentences or kill themselves.

tgstation-server committed:
Automatic changelog generation for PR #67796 [ci skip]

ShizCalev committed:
Fixes some door shocking issues (#68278)

* Fixes some door shocking issues

tgstation-server committed:
Automatic changelog generation for PR #68278 [ci skip]

orthography committed:
Asimov++ made editable identically to Asimov (#68295)

Co-authored-by: Mothblocks <35135081+Mothblocks@users.noreply.github.com>

tgstation-server committed:
Automatic changelog generation for PR #68295 [ci skip]

Thunder12345 committed:
Revert "Fixes admin.fun list" (#68369)

Revert "Fixes admin.fun list (#68336)"

This reverts commit c17acd07e64a72453cd85b99cd3e1d9a080b78fc.

CoffeeDragon16 committed:
Golems now have tongues again, except cardboard golems (#68168)

* vocal cord slot

* puts vocal chords back

* reverts changes i couldnt figure out how to make mutant_vocalcords work

* cleanup

* woops

* Makes cardboard golems have no tongues

* adds no tongue to the greet

tgstation-server committed:
Automatic changelog generation for PR #68168 [ci skip]

Sucy-Manbavaran committed:
Bell pepper resprite (#68283)

imageadd: added new Bell Pepper, Raw Stuffed Pepper, Stuffed Pepper and Grilled Bell Peppers Sprites

Pickle-Coding committed:
[NO GBP]You no longer need to change 2 other defines based on math in order to change one of the powerloss defines. (#67982)

Adds powerloss_linear_offset constant, which gets calculated to offset the linear function to allow the transition between the cubic and linear powerloss functions to be differentiable.

actions-user committed:
Automatic changelog compile [ci skip]

ZephyrTFA committed:
Resolve incorrect paths in the header comment for tgui chat styling (#68356)

* fix the paths for the chat style sheets

* Update interface/stylesheet.dm

Co-authored-by: Aleksej Komarov <stylemistake@gmail.com>

Christmas5 committed:
Cleaning up changelings a little. (#68117)

More aspects will now be affected by shapeshifting, creating a more accurate disguise:

    Age.
    Physique.
    Hair gradients.
    Neck items.
    Underwear colour.

Last resort will now respawn you as a /human rather than /human/species/monkey.
As the egg hatches, the game will spawn a human and turn them into a monkey, using code similar to that of lesser form (instead of spawning an actual monkey). This should prevent issues with randomised names.

Using Human form after Lesser form should now shapeshift you into the correct species.
Previously it just ran /humanize() without any argument, which means it defaulted to turning you into a human. This should not be an issue, considering that it still runs a regular shapeshift afterwards and updates your DNA, but for some reason that's not enough.

The Human form ability now uses a radial menu instead of a list.
De-clutters the code a little by reusing the code for the radial menu that's used when you shapeshift into someone. It also makes the ability smoother to use and more consistent.

Lesser form now clears your flesh disguise as it transforms you.

tgstation-server committed:
Automatic changelog generation for PR #68117 [ci skip]

necromanceranne committed:
Touches up the Makarov sprites one last time, totally doesn't add a shadowy fake 10mm Makarov. (#68160)

Redoes the Makarov sprite again, adds the Ansem pistol, the shadowy fake of the Makarov in 10mm.

tgstation-server committed:
Automatic changelog generation for PR #68160 [ci skip]

zxaber committed:
New Sprite for the Airlock Shock Indicator as seen by Diagnostics Huds (#68121)

Now with 50% less ambiguity

tgstation-server committed:
Automatic changelog generation for PR #68121 [ci skip]

MidoriWroth committed:
fixing the merge conflict i hope

MidoriWroth committed:
ok maybe this will fix it part 1

MidoriWroth committed:
please work part 2

MidoriWroth committed:
once more

MidoriWroth committed:
test 1

MidoriWroth committed:
test 2

MidoriWroth committed:
ok final test before i get help

MidoriWroth committed:
Merge branch 'FoodExpansion3'

MidoriWroth committed:
Revert "Merge branch 'FoodExpansion3'"

This reverts commit 2256e1a0afeffc12895aa7eea1f9f2ea640edc71, reversing
changes made to 120ecddbda6d1e8b114e0f686684a6b70a997ec9.

MidoriWroth commented:
I fucked this up real bad lmao, just gonna start over

Sucy-Manbavaran commented:
> I fucked this up real bad lmao, just gonna start over

godspeed, i'll keep in mind NOT to base any food related coding i do off your code 😆
